doc/release/upcoming_changes/16841.changes.rst

```diff
@@ -0,0 +1,5 @@
+`np.linspace` on integers now use floor
+---------------------------------------
+When using a `int` dtype in `np.linspace`, previously float values would
+be rounded towards zero. Now `np.floor` is used instead, which rounds toward
+``-inf``. This changes the results for negative values.
```

Comment:
@eric-wieser does this capture the idea better?

Reply:
Needs to be `numpy.linspace` for the rst refs to work, right?

Otherwise looks good - I think we might want a `..versionchanged` in the docs too, and maybe an example of before / after here.